### ISSUE TITLE
MedicinalProductDefinition with multiple Ingredients.

### ISSUE THREAD

**Issue by a-gkogkidis** [hl7-eu/unicom-ig#111]
How can I know beforehand if MPD has multiple Ingredients without using the string of the substances that are included there? Maybe a counter that this MPD has 3 ingredients or more?

I know that there is no direct link from MPD to Ingredient, rather the oposite Ingredient has reference to MPD.

**Comment by a-gkogkidis** [comment]
Or the other thing that might be more helpful to is to know if an Ingredient is part of a MPD with other Ingredients. Since the query starts from Ingredient. Moreover, having this knowledge in the resource could be heplful at some point when we will start testing substitution for multiple ingredients, as right now the testing is only to MPDs with one ingredient.

**Comment by joofio** [comment]
I guess Jose already stated this, but adding an concrete example:
http://fhir.hl7.pt:8080/fhir/Ingredient?for=MedicinalProductDefinition/mp-2516&_total=accurate
total returns 2: so more than one.

Is this enough?

**Comment by rlindstrm** [comment]
I don't think this is what Argiris wants, @joofio . You requested all Ingredients of one product. It's helpful but not for this use case.

Argiris's big goal is to get all products that have, for example, exactly 0.02mg ethinylestradiol + 3mg drosperinone ([this ](https://build.fhir.org/ig/hl7-eu/unicom-ig/branches/mpd-r5/Bundle-006-Yaz-002mg3mg-tablet-SE-FullProduct.html) and [this](https://build.fhir.org/ig/hl7-eu/unicom-ig/branches/mpd-r5/Bundle-007-Jangee-002mg-3mg-Film-coated-tablet-EE-FullProduct.html)). 
Or to get all products that have 10mg amlodipine (but NOT get the ones that have [10mg amlodipine + 160mg valsartan + 12.5mg Hydrochlorothiazide](https://build.fhir.org/ig/hl7-eu/unicom-ig/branches/mpd-r5/Bundle-015-CopaliaHCT-EE-FullProduct.html)).

I am quite sure José has not started working on it. And it's not very straightforward, especially when you add the fact, that you might not know if the substance given on an eP is in the strength or reference strength section in the data.

**Comment by joofio** [comment]
Yes I understand that, but doing that in one go, with vanilla search, i am not seeing how to do it.

The example I gave is that you know that MedicinalProductDefinition/mp-2516 has more than 1 ingredient since the return is "total=2"

**Comment by rlindstrm** [comment]
I don't think we're even dreaming of doing it in one search. And I did not realise the count part in your search - that could be helpful, yes.

**Comment by joofio** [comment]
in here we want (writting it down so i can think of how to do it):

getting the medicinal product that is referenced by two specific ingredients in the "for" element

**Comment by joofio** [comment]
http://fhir.hl7.pt:8080/fhir/MedicinalProductDefinition?_has:Ingredient:for:identifier=059QF0KO0R&_has:Ingredient:for:identifier=GGO4Y809LO

would this work?

**Comment by joofio** [comment]
this query gets all MedicinalProductDefinition that are referenced in the "for" element of the ingredient with these identifiers

**Comment by joofio** [comment]
something like this for the products at hand?
http://fhir.hl7.pt:8080/fhir/MedicinalProductDefinition?_has:Ingredient:for:strength-ratio-and-substance=3$https://spor.ema.europa.eu/v2/SubstanceDefinition|100000092375&_has:Ingredient:for:strength-ratio-and-substance=0.02$https://spor.ema.europa.eu/v2/SubstanceDefinition|100000091531
